src/app/screens/details_actions.rs

```diff
@@ -295,40 +299,91 @@ impl DetailsActions {
             .arg("-C")
             .arg(kernel_tree.path())
             .arg("rev-parse")
             .arg("--abbrev-ref")
             .arg("HEAD")
             .output()
-            .unwrap();
-        let mut original_branch = String::from_utf8_lossy(&original_branch.stdout).to_string();
-        original_branch.pop();
-        let _ = Command::new("git")
+            .map_err(|e| format!("failed to get current branch: {}", e))?;
+
+        let mut branch = String::from_utf8_lossy(&original_branch.stdout).to_string();
+        branch.pop();
+        Ok(branch)
+    }
+
+    /// Switch the supplied kernel tree to the supplied branch, if it exists.
+    ///
+    /// Returns `()` on sucess and a `String` with the error message on failure.
+    fn switch_to_branch(&self, kernel_tree: &KernelTree, branch: &str) -> Result<(), String> {
+        let output = Command::new("git")
             .arg("-C")
             .arg(kernel_tree.path())
             .arg("switch")
-            .arg(kernel_tree.branch())
+            .arg(branch)
             .output()
-            .unwrap();
+            .map_err(|e| format!("failed to switch branch: {}", e))?;
+
+        if !output.status.success() {
+            return Err(format!(
+                "failed to switch to branch '{}': {}",
+                branch,
+                String::from_utf8_lossy(&output.stderr)
+            ));
+        }
+
+        Ok(())
+    }
+
+    /// Create a new branch suffixed with the current timestamp.
+    ///
+    /// Returns a `String` with the branch name on success or the
+    /// error message on failure.
+    fn create_target_branch(
+        &self,
+        kernel_tree: &KernelTree,
+        config: &Config,
+    ) -> Result<String, String> {
+        self.switch_to_branch(kernel_tree, &kernel_tree.branch())?;
+
         let target_branch_name = format!(
             "{}{}",
             config.git_am_branch_prefix(),
             chrono::Utc::now().format("%Y-%m-%d-%H-%M-%S")
         );
-        let _ = Command::new("git")
+
+        let output = Command::new("git")
             .arg("-C")
             .arg(kernel_tree.path())
             .arg("checkout")
             .arg("-b")
             .arg(&target_branch_name)
             .output()
-            .unwrap();
+            .map_err(|e| format!("failed to create target branch: {}", e))?;
+
+        if !output.status.success() {
+            return Err(format!(
+                "failed to create branch '{}': {}",
+                target_branch_name,
+                String::from_utf8_lossy(&output.stderr)
+            ));
+        }
+
+        Ok(target_branch_name)
+    }
 
-        // 8. Apply the patchset
+    /// Apply the selected patchset on the given `kernel_tree` with arguments from `Config`
+    ///
+    /// Returns `()` on sucess and a `String` containing the error message on failure.
+    fn apply_patches(&self, kernel_tree: &KernelTree, config: &Config) -> Result<(), String> {
```

Comment:
I think we can enhance the name of this function to not be mistaken with `apply_patchset` and be more specific. Something like

```suggestion
    fn run_git_am(&self, kernel_tree: &KernelTree, config: &Config) -> Result<(), String> {
```